common/djangoapps/pipeline_mako/__init__.py

```diff
@@ -79,3 +82,78 @@ def render_individual_js(package, paths, templates=None):
+
+
+def render_require_js_path_overrides(path_overrides):  # pylint: disable=invalid-name
+    """Render JavaScript to override default RequireJS paths.
+
+    The Django pipeline appends a hash to JavaScript files,
+    so if the JS asset isn't included in the bundle for the page,
+    we need to tell RequireJS where to look.
+
+    For example:
+
+        "js/vendor/jquery.min.js" --> "js/vendor/jquery.min.abcd1234"
+
+    We would then add a line in a <script> tag:
+
+        require.paths['jquery'] = 'js/vendor/jquery.min.abcd1234'
+
+    so that any reference to 'jquery' in a JavaScript module
+    will cause RequireJS to load '/static/js/vendor/jquery.min.abcd1234.js'
+
+    If running in DEBUG mode (as in devstack), the resolved JavaScript URLs
+    won't contain hashes, so the new paths will match the original paths.
+
+    Arguments:
+        path_overrides (dict): Mapping of RequireJS module names to
+            filesystem paths.
+
+    Returns:
+        unicode: The HTML of the <script> tag with the path overrides.
+
+    """
+    # Render the <script> tag that overrides the paths defined in `require.paths`
+    # Note: We don't use a Mako template to render this because Mako apparently
+    # acquires a lock when loading templates, which can lead to a deadlock if
+    # this function is called from within another template.
+    html = ['<script type="text/javascript">']
+
+    # The rendered <script> tag with overrides should be included *after*
+    # the application's RequireJS config, which defines a `require` object.
+    # Just in case the `require` object hasn't been loaded, we create a default
+    # object.  This will avoid a JavaScript error that might cause the rest of the
+    # page to fail; however, it may mean that these overrides won't be available
+    # to RequireJS.
+    html.extend([
+        'var require = require || RequireJS.require;',
+        # 'require.paths = require.paths || [];'
+    ])
+
+    # Specify override the base URL to point to STATIC_URL
+    # html.append(
+        # "require.baseUrl = '{url}'".format(
+            # url=django_settings.STATIC_URL
+        # )
+    # )
+
+    for module, url_path in path_overrides.iteritems():
+        # Calculate the full URL, including any hashes added to the filename by the pipeline.
+        # This will also include the base static URL (for example, "/static/") and the
+        # ".js" extension.
+        actual_url = staticfiles_storage.url(url_path)
+
+        # RequireJS assumes that every file it tries to load has a ".js" extension, so
+        # we need to remove ".js" from the module path.
+        # RequireJS also already has a base URL set to the base static URL, so we can remove that.
+        path = actual_url.replace('.js', '').replace(django_settings.STATIC_URL, '')
+
+        # Add the path override to the inline JavaScript.
+        html.append(
+            "require.s.contexts._.config.paths['{module}'] = '{path}';".format(
```

Comment:
@andy-armstrong Here are more details. We are trying to update `require.paths` with new paths but this was not working and requirejs is loading unhashed versions of js files. This is because `paths` attribute is not assigned and accessible directly on requirejs. So i found the above ugly [way](https://groups.google.com/forum/#!topic/requirejs/Hf-qNmM0ceI) to update the paths dynamically. Now hashed js files are being loaded but i don't think this solution is the right way to solve this issue.